### ISSUE TITLE
Using possibly undefined loop variable

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#1081]
In GitLab by @schabot on Mar 8, 2019, 03:33

Hello,

*Please describe how you installed Flake8*

```bash
$ python3 -m venv env
$ source env/bin/activate
$ pip install flake8
```

*Please provide the exact, unmodified output of `flake8 --bug-report`*

```
{
  "dependencies": [
    {
      "dependency": "entrypoints",
      "version": "0.3"
    }
  ],
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.7.2+",
    "system": "Linux"
  },
  "plugins": [
    {
      "is_local": false,
      "plugin": "mccabe",
      "version": "0.6.1"
    },
    {
      "is_local": false,
      "plugin": "pycodestyle",
      "version": "2.5.0"
    },
    {
      "is_local": false,
      "plugin": "pyflakes",
      "version": "2.1.0"
    }
  ],
  "version": "3.7.4"
}
```

*Please describe the problem or feature*

In the given script (see below), the variable `value` is possibly undefined, but it is not reported by flake8.

Flake8 should give a warning about that (W0631 is given by Pylint).

*If this is a bug report, please explain with examples (and example code) what you expected to happen and what actually happened.*

```python
#!/usr/bin/env python3
# coding: utf-8

"""
    the variable `value` is possibly undefined.
"""


def print_content(iterable):
    """
        print the content of the given iterable
    """
    for value in iterable:
        print(value)
    last_value = value
    print('The last_value is ', last_value)


print_content('abcd')
print_content([])
```

**Comment by asottile** [comment]
In GitLab by @asottile on Mar 8, 2019, 03:58

Flake8 is merely a framework which other linters run under and does not implement any rules itself. I'd suggest either writing a plugin or reporting this to pyflakes / pycodestyle. (though, this is probably too-high false-positive for pyflakes and not really in the style of pycodestyle -- it probably belongs in pylint)

**Comment by asottile** [comment]
In GitLab by @asottile on Mar 8, 2019, 03:58

closed
